Simplenote/ShortcutHandler.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  ShortcutHandler.swift
+//  Simplenote
+//
+//  Created by Charlie Scheer on 5/23/24.
+//  Copyright © 2024 Simperium. All rights reserved.
+//
+
+import Foundation
+
+class ShortcutsHandler: NSObject {
+    @objc
+    static var shared = ShortcutsHandler()
+
+    /// Is User authenticated?
+    ///
+    private var isAuthenticated: Bool {
+        return SimplenoteAppDelegate.shared().simperium.user?.authenticated() == true
+    }
+
+    /// Handles a UserActivity instance. Returns true on success.
+    ///
+    @objc
+    func handleUserActivity(_ userActivity: NSUserActivity) -> Bool {
+        guard let type = ActivityType(rawValue: userActivity.activityType),
+              isAuthenticated else {
+            return false
+        }
+
+        switch type {
+        case .newNoteShortcut:
+            SimplenoteAppDelegate.shared().noteEditorViewController.createNote(from: nil)
+        }
+
+        return true
+    }
```

Comment:
Suggestion!!

I've noticed the `SimplenoteAppDelegate` builds `ShortcutHandler`, and then `ShortcutHandler` access the `SimplenoteAppDelegate` singleton to check for the authentication state, **and then** to create the note.

Both entities are tightly coupled. A cleaner approach could involve:

1. Dropping the `shared` property (the singleton!)
2. Building a protocol that gets you what you need from the AppDelegate (`isAuthenticated`?)
3. Building the Handler, and passing the reference of the "AuthenticationStatusProvider"
4. Making the AppDelegate conform to this protocol

**Alternatively** (and I'd probably go for this one): all of the `Handler` code could just live in  `SimplenoteAppDelegate+Shortcuts`

Reply:
Good call.  I was copying how it is setup in iOS, but there is a lot more going on with the shortcut handler in that app.  I have moved all of it into the app delegate since it is so simple